x-pack/platform/plugins/shared/alerting/server/routes/maintenance_window/apis/external/delete/delete_maintenance_window_route.test.ts

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { httpServiceMock } from '@kbn/core/server/mocks';
+import { licenseStateMock } from '../../../../../lib/license_state.mock';
+import { verifyApiAccess } from '../../../../../lib/license_api_access';
+import { mockHandlerArguments } from '../../../../_mock_handler_arguments';
+import { maintenanceWindowClientMock } from '../../../../../maintenance_window_client.mock';
+import { deleteMaintenanceWindowRoute } from './delete_maintenance_window_route';
+import { getMockMaintenanceWindow } from '../../../../../data/maintenance_window/test_helpers';
+import { MaintenanceWindowStatus } from '../../../../../../common';
+
+const maintenanceWindowClient = maintenanceWindowClientMock.create();
+
+jest.mock('../../../../../lib/license_api_access', () => ({
+  verifyApiAccess: jest.fn(),
+}));
+
+const mockMaintenanceWindow = {
+  ...getMockMaintenanceWindow(),
+  startDate: new Date().toISOString(),
+  endDate: new Date().toISOString(),
+  status: MaintenanceWindowStatus.Running,
+  id: 'test-id',
+};
+
+describe('deleteMaintenanceWindowRoute', () => {
+  beforeEach(() => {
+    jest.resetAllMocks();
+  });
+
+  test('should delete the maintenance window', async () => {
+    const licenseState = licenseStateMock.create();
+    const router = httpServiceMock.createRouter();
+
+    deleteMaintenanceWindowRoute(router, licenseState);
+
+    maintenanceWindowClient.delete.mockResolvedValueOnce(mockMaintenanceWindow);
+    const [config, handler] = router.delete.mock.calls[0];
+    const [context, req, res] = mockHandlerArguments(
+      { maintenanceWindowClient },
+      { params: { id: 'test-id' } }
+    );
+
+    expect(config.path).toEqual('/api/maintenance_window/{id}');
+    expect(config.options).toMatchInlineSnapshot(`
+      Object {
+        "access": "public",
+        "summary": "Deletes a maintenance window by ID.",
```

Comment:
Unless there's another endpoint that deletes a single maintenance window by another type of identifier, this can be shortened:
```suggestion
        "summary": "Delete a maintenance window.",
```